protostar/commands/test/test_case_runners/test_case_runner_factory.py

```diff
@@ -29,15 +26,16 @@ def make(self, test_case: TestCase) -> TestCaseRunner:
         )
 
         if state.config.mode is TestMode.FUZZ:
-            return FuzzTestCaseRunner(
-                fuzz_test_execution_environment=FuzzTestExecutionEnvironment(state),
+            return TestCaseRunner(
+                execution_environment=FuzzTestExecutionEnvironment(state),
                 test_case=test_case,
                 output_recorder=state.output_recorder,
+                test_case_result_decorator=FuzzTestCaseResultDecorator(),
             )
```

Comment:
`TestCaseRunner` isn't designed well. It allows providing incompatible environments and decorators. Can you constrain it somehow with generics? The previous solution didn't allow that.

IMO, Template method pattern was better choice than decorators.

Reply:
> IMO, Template method pattern was better choice than decorators.

Don't decorators inheritance realize exactly the same pattern, but decorators themselves are encapsulated via a strategy pattern now?

Reply:
this comment is no more relevant